### ISSUE TITLE
Migrate to typescript and Promise

### ISSUE THREAD

**Issue by botika** [BryanDonovan/node-cache-manager#198]
It's been a long time since node has Promise support. Can I migrate the project to typescript and Promise and do the PR?

**Comment by BryanDonovan** [comment]
Hi.  There is Promise support.  Maybe it's missing somewhere though?

**Comment by botika** [comment]
I've already seen it, but the internal structure still uses callbacks in all functions. To implement a cache with node redis v4, which is based on promises, I have to use [callbackify](https://nodejs.org/api/util.html#util_util_callbackify_original). The idea is that everything is based on Promises and remove the structure of callbacks.

**Comment by BryanDonovan** [comment]
It seems like this would be a large undertaking and would not be compatible with the existing cache stores.

**Comment by botika** [comment]
Yes, of course they should be updated. But with a mayor version that contemplates the changes. And migrating from callback to Promise is really easy in the 7 methods that the Store has. But above all it would be faster since it would not have to check the arguments. I think it will be a necessary step, sooner or later. And my redis store would be much simpler to implement. Up to you.

**Comment by botika** [comment]
I have made this version https://github.com/botika/node-cache-manager-redis-store, if you want me to do the PR say so. Otherwise feel free to close this issue. Thanks.

**Comment by BryanDonovan** [comment]
Sorry for the delay.  I was on vacation.  

I would be ok with either of the following options:

1. You  migrate this project to use Promises, but *not* TypeScript.  I don't use TypeScript and have no intention of learning it.
2. You (or someone else) takes over ownership of the entire node-cache-manager project and do with it as you see fit.

What do you think?

**Comment by botika** [comment]
I would love to take over.

**Comment by botika** [comment]
I have accepted the invitation from npm. I have made a github team and I have invited you to move the repo there. https://github.com/node-cache-manager

**Comment by BryanDonovan** [comment]
Thanks, I'll do this as soon as I can.

**Comment by BryanDonovan** [comment]
I've moved the repo over to the new `node-cache-manager` team.

**Comment by botika** [comment]
Ok, I'll get to work.
